### PR TITLE
Introducing ExposedDatetimeMixin

### PR DESCRIPTION
**Pull request by Thingus**
Closes issue #4497

### I have:

- [x] Formatted any Python files with [black](https://github.com/ambv/black)
- [x] Brought the branch up to date with master
- [x] Added any relevant Github labels
- [x] Added tests for any new additions
- [x] Added or updated any relevant documentation
- [ ] Added an Architectural Decision Record (ADR), if appropriate
- [x] Added an [MPLv2 License Header](https://www.mozilla.org/en-US/MPL/headers/) if appropriate
- [x] Updated the [Changelog](https://github.com/Flowminder/FlowKit/blob/master/CHANGELOG.md) 

### Description

Creates a mixin that exposes start_date and end_date internally as datetimes, but externally as strings

### CODE REVIEW

File: flowmachine/tests/test_exposed_datetime_mixin.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+from datetime import datetime
+
+
+class DummyDatetime(ExposedDatetimeMixin):
+    def __init__(self, start_date, end_date):
+        self.start_date = start_date
+        self.end_date = end_date
+
+
+def test_exposed_datetime_mixin():
```
Comment (jc-harrison) [inline flowmachine/tests/test_exposed_datetime_mixin.py:14]: Might be good to parametrise this test, to test that passing `datetime.date` or `datetime.datetime` arguments has the same outcome.

File: flowmachine/flowmachine/core/mixins/exposed_datetime_mixin.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    a class as datetime objects (`_start_dt` and `_end_dt` respectively), but externally as strings.
+    """
+
+    @property
+    def start_date(self):
+        return standardise_date(self._start_dt)
+
+    @start_date.setter
+    def start_date(self, value):
+        self._start_dt = datetime.strptime(standardise_date(value), "%Y-%m-%d %H:%M:%S")
```
Comment (greenape) [inline flowmachine/flowmachine/core/mixins/exposed_datetime_mixin.py:25]: Trivial, but could whack that in as a comparison utils func `standardise_date_to_datetime`?

**Review by greenape — APPROVED** [review]
I'm happy with this modulo one minor suggestion. Next job would be to apply it to the 40-odd query classes that would benefit.

Probably worth adding a changelog entry for it, cos extending fm-library for specific purpose analysis is a valid use case.

File: flowmachine/flowmachine/core/mixins/exposed_datetime_mixin.py (hunk 2 above)
Comment (Thingus) [reply flowmachine/flowmachine/core/mixins/exposed_datetime_mixin.py:25]: Worth renaming `standardise_date` to `standardise_date_to_string` at a later date?

**Comment by Thingus** [comment]
I'll stick in an issue to add this in

File: flowmachine/flowmachine/core/mixins/exposed_datetime_mixin.py (hunk 2 above)
Comment (Thingus) [reply flowmachine/flowmachine/core/mixins/exposed_datetime_mixin.py:25]: > I'm happy with this modulo one minor suggestion. Next job would be to apply it to the 40-odd query classes that would benefit.

I'll open p-later issue for this

**Comment by Thingus** [comment]
OK, I think this is ready to merge, pending CI.

**Pull request merged**